### PR TITLE
Add new article: Is Cockpit Secure?

### PR DESCRIPTION
**Pull request by stefwalter**
This is an article I wrote a while back. I just touched it up for inclusion on this site.

### CODE REVIEW

File: _posts/2018-09-13-is-cockpit-secure.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+[SELinux](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Security-Enhanced_Linux/sect-Security-Enhanced_Linux-Confining_Users-Confining_Existing_Linux_Users_semanage_login.html)
+ and other senses.
+The logged in browser and the javascript code it is running can interact directly
+with the system as part of that session.
+
+Try this out if you want. [Log into Cockpit](http://cockpit-project.org/running.html),
+and pop up the Javascript console by pressing *Ctrl-Shift-J* in your browser. Now type
+commands like the following:
+
+    > zz = cockpit.spawn(["id"])
```
Comment (larskarlitski) [inline _posts/2018-09-13-is-cockpit-secure.md]: This only returns a promise now, doesn't it? How about

```
zz = cockpit.spawn(["id"]).then(console.log)
```

File: _posts/2018-09-13-is-cockpit-secure.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+tools used by your Linux distribution.
+
+If your security requirements are strict and you must avoid browsers at all
+costs, then be sure to audit all possible places that browsers were involved
+in what's running on your systems. It's harder than you think.
+
+Unless your system is very isolated, it turns out that browsers are
+likely already highly involved in the security paths for your systems.
+Cockpit does not change this situation significantly.
+
```
Comment (larskarlitski) [inline _posts/2018-09-13-is-cockpit-secure.md]: Does it make sense to mention the possibility of using a bastion host here?

File: _posts/2018-09-13-is-cockpit-secure.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+Back to Cockpit. What does that session look like? Cockpit itself has no
+special privileges. The credentials of the logged in user start a login
+session, and Cockpit can perform exactly the tasks that the logged in user
+has access to. It has no more or less permissions.
+
+You can examine anything about Cockpit security. Log into Cockpit, open the
+*Terminal* page and run commands like the following:
+
+    :::text
```
Comment (larskarlitski) [inline _posts/2018-09-13-is-cockpit-secure.md]: Is this a marker for other markup? Indenting code blocks is enough in Markdown. (These also appear further down.)

File: _posts/2018-09-13-is-cockpit-secure.md [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+Although individual parts of Cockpit can
+[override this default](http://cockpit-project.org/guide/latest/packages.html#package-manifest),
+it's rarely done. The default security policy looks like this:
+
+     Content-Security-Policy: default-src 'self' connect-src 'self' ws: wss:
+
+A failure of *Content Security Policy* will look something like this in your browser's
+javascript console:
+
+![Content Security Policy denial](/images/csp-denied.png)
```
Comment (larskarlitski) [inline _posts/2018-09-13-is-cockpit-secure.md]: This image is missing.

File: _posts/2018-09-13-is-cockpit-secure.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+When you're trying this out in real life, specify an alternate server on the Cockpit
+login screen. The SSH protocol will be used to connect to it:
+
+![Login Alternate Server](/images/cockpit-login-alternate.png)
+
+Or you can add other machines to a local dashboard, and Cockpit will connect to
+them via SSH. Even usage of SSH key based authentication works great:
+
+<iframe width="853" height="480" src="https://www.youtube.com/embed/Ye5YlVNXC9w" frameborder="0" allowfullscreen></iframe>
+
```
Comment (larskarlitski) [inline _posts/2018-09-13-is-cockpit-secure.md]: The video is not rendered.

File: _posts/2018-09-13-is-cockpit-secure.md [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+still allows you and your systems to accomplish what they need to do.
+
+Here's info about [Cockpit's](http://cockpit-project.org/) security, to help
+you make those choices. You'll find not only has Cockpit got a solid security
+story, but you can use it in all sorts of different ways depending on what
+kind of security your systems need.
+
+## Cockpit is a Linux session in your browser
+
+Let start by stacking the deck against Cockpit. First off it's important to
```
[the remark this answers is not in the document]
Comment (stefwalter) [reply _posts/2018-09-13-is-cockpit-secure.md]: Fixed.

File: _posts/2018-09-13-is-cockpit-secure.md (hunk 1 above)
Comment (stefwalter) [reply _posts/2018-09-13-is-cockpit-secure.md]: Assigning to zz does all that magic. Try it.

File: _posts/2018-09-13-is-cockpit-secure.md (hunk 2 above)
Comment (stefwalter) [reply _posts/2018-09-13-is-cockpit-secure.md]: This security consideration has nothing to do with bastion hosts. Your browser is your browser. But we do mention bastion hosts below.

File: _posts/2018-09-13-is-cockpit-secure.md (hunk 3 above)
Comment (stefwalter) [reply _posts/2018-09-13-is-cockpit-secure.md]: removed

File: _posts/2018-09-13-is-cockpit-secure.md (hunk 4 above)
Comment (stefwalter) [reply _posts/2018-09-13-is-cockpit-secure.md]: Fixed.

File: _posts/2018-09-13-is-cockpit-secure.md (hunk 5 above)
Comment (stefwalter) [reply _posts/2018-09-13-is-cockpit-secure.md]: Strange it looks just like all the other posts.

Comment (larskarlitski) [reply _posts/2018-09-13-is-cockpit-secure.md]: I was looking at the rendered output here by clicking "View" at the top of the file. That probably just doesn't allow arbitrary html.

**Comment by larskarlitski** [comment]
Thanks!

File: _posts/2018-09-13-is-cockpit-secure.md [hunk 7]
```diff
@@ -0,0 +1,255 @@
+---
+title: Is Cockpit Secure?
+date: 2018-09-13 10:00
+category: blog
+tags: cockpit
+slug: is-cockpit-secure
+---
+
+Security is of your Linux systems vital. It's not a binary thing though.
```
Comment (davidmalcolm) [inline _posts/2018-09-13-is-cockpit-secure.md]: The wording of the first sentence seems strange.  Perhaps:

  "Security is vital to your Linux systems."

instead?

Comment (martinpitt) [reply _posts/2018-09-13-is-cockpit-secure.md]: The Yoda talk fixed I have. Thanks for spotting!

**Pull request merged**